leaf-ui/rappid-extensions/AnalysisInspector.js

```diff
@@ -154,7 +86,7 @@ var AnalysisInspector = Backbone.View.extend({
 		$('#abs-time-pts').val(analysisRequest.absTimePts);
 
 		// This is needed to allow the user to delete relative assignments
-		this.setDeleteRelAssignmentListener();
+		//this.setDeleteRelAssignmentListener();
```

Comment:
Can you help me understand what this is for? Sorry I'm dumb

Reply:
Yes so as far as I understand this render function is basically passed a template and the script to go along with it. Because of the way the Relative Intention Assignments table is added to (basically adding more html strings to the table to make new rows, instead of being rendered when the table is initially created) it seems like this listener was previously necessary so that the code would be "waiting" at all times to see if the delete button was clicked on any added rows. 
Basically: In AnalysisInspector the html elements in the template are tied to functions and then all rendered together. You can see this inside the events portion of the AnalysisInspector variable. This means that any added html would not be dynamic unless a listener function was called inside the render. In the toolbar, the render is removed and the html is tied directly event calls. This meant I could delete the setDeleteRelAssignmentListener() in AnalysisInspector and in onFunctions.js I could just tie the delete action to the html directly, removing the need for the listener at any point. 

This is just my understanding so far and there are definitely holes in my knowledge on this as I have very little to no previous javascript experience, but as far as I have seen and tested this is the case and the listener is no longer needed here or in onFunctions.js